api/src/main/java/org/ehcache/spi/service/OptionalServiceDependencies.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi.service;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that allows a {@link Service} implementation to declare an optional dependency on other {@code Service}s.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface OptionalServiceDependencies {
+
+  /**
+   * Array of {@link Service} dependency classes
+   */
+  String[] value();
```

Comment:
That would make the class using this annotation throw `ClassNotFoundException` at runtime which would be counter-productive, wouldn't it?

Reply:
I'm not sure... I'm *hoping* that this might 'do the right thing' in as similar manner to how if an annotation class is missing at runtime that it just fails to load.